tmt/steps/discover/__init__.py

```diff
@@ -131,6 +132,37 @@ def log_import_plan_details(self) -> None:
             for key, value in cast(dict[str, str], remote_plan_id.to_minimal_spec()).items():
                 self.verbose(f'import {key}', value, 'green')
 
+    def filter_for_rerun(self) -> None:
+        """ Filter out passed tests from previous run data """
+        assert isinstance(self.step.parent, tmt.base.Plan)  # narrow type
+        old_results: Path = self.step.parent.last_run_execute / 'results.yaml'
+        results = [
+            Result.from_serialized(data) for data in
+            tmt.utils.yaml_to_list(self.read(old_results))]
+        results_failed: list[str] = []
+        results_passed: list[Result] = []
+        for result in results:
+            if (
+                    result.result is not tmt.result.ResultOutcome.PASS and
+                    result.result is not tmt.result.ResultOutcome.INFO):
+                results_failed.append(result.name)
+            else:
+                results_passed.append(result)
+
+        # Overwrite previous run results to only include passed cases
```

Comment:
I don't understand this move. Can the original run remain as it was, unmodified? Its `log.txt` would tell one story, its `results.yaml` another one. Is there a technical reason to modify it?

Reply:
Ok, I guess it's to have just the positive results to be merged with the subset we get from the re-run. If that's the case, I'd prefer to save the reduced `results.yaml` in the new run rather than changing the original. I.e. something like `old_results = self.step.workdir / 'original-results-positive-ones-only.yaml'` (ignore the actual filename, for the same of an example :)